protocol/rest/rest_server/go_restful_server.go

```diff
@@ -0,0 +1,310 @@
+	grs.container = restful.NewContainer()
+	grs.srv = &http.Server{
+		Handler: grs.container,
+	}
+	ln, err := net.Listen("tcp", url.Location)
+	if err != nil {
+		panic(perrors.New(fmt.Sprintf("Restful Server start error:%v", err)))
+	}
+
+	go func() {
+		err := grs.srv.Serve(ln)
+		if err != nil && err != http.ErrServerClosed {
+			logger.Errorf("[Go Restful] http.server.Serve(addr{%s}) = err{%+v}", url.Location, err)
+		}
+	}()
+}
+
+func (grs *GoRestfulServer) Deploy(invoker protocol.Invoker, restMethodConfig map[string]*rest_interface.RestMethodConfig) {
+	svc := common.ServiceMap.GetService(invoker.GetUrl().Protocol, strings.TrimPrefix(invoker.GetUrl().Path, "/"))
+	for methodName, config := range restMethodConfig {
+		// get method
+		method := svc.Method()[methodName]
+		argsTypes := method.ArgsType()
+		replyType := method.ReplyType()
+		ws := new(restful.WebService)
+		ws.Path(config.Path).
+			Produces(strings.Split(config.Produces, ",")...).
+			Consumes(strings.Split(config.Consumes, ",")...).
+			Route(ws.Method(config.MethodType).To(getFunc(methodName, invoker, argsTypes, replyType, config)))
+		grs.container.Add(ws)
+	}
+
+}
+
+func getFunc(methodName string, invoker protocol.Invoker, argsTypes []reflect.Type, replyType reflect.Type, config *rest_interface.RestMethodConfig) func(req *restful.Request, resp *restful.Response) {
+	return func(req *restful.Request, resp *restful.Response) {
+		var (
+			err  error
+			args []interface{}
+		)
+		if (len(argsTypes) == 1 || len(argsTypes) == 2 && replyType == nil) && argsTypes[0].String() == "[]interface {}" {
+			args = getArgsInterfaceFromRequest(req, config)
+		} else {
+			args = getArgsFromRequest(req, argsTypes, config)
+		}
+		result := invoker.Invoke(context.Background(), invocation.NewRPCInvocation(methodName, args, make(map[string]string, 0)))
+		if result.Error() != nil {
+			err = resp.WriteError(http.StatusInternalServerError, result.Error())
+			if err != nil {
+				logger.Errorf("[Go Restful] WriteError error:%v", err)
+			}
+			return
+		}
+		err = resp.WriteEntity(result.Result())
+		if err != nil {
+			logger.Error("[Go Restful] WriteEntity error:%v", err)
+		}
+	}
+}
+func (grs *GoRestfulServer) UnDeploy(restMethodConfig map[string]*rest_interface.RestMethodConfig) {
+	for _, config := range restMethodConfig {
+		ws := new(restful.WebService)
+		ws.Path(config.Path)
+		err := grs.container.Remove(ws)
+		if err != nil {
+			logger.Warnf("[Go restful] Remove web service error:%v", err)
+		}
+	}
+}
+
+func (grs *GoRestfulServer) Destroy() {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	if err := grs.srv.Shutdown(ctx); err != nil {
+		logger.Errorf("[Go Restful] Server Shutdown:", err)
+	}
+	logger.Infof("[Go Restful] Server exiting")
+}
+
+func getArgsInterfaceFromRequest(req *restful.Request, config *rest_interface.RestMethodConfig) []interface{} {
+	argsMap := make(map[int]interface{})
```

Comment:
@Patrick0308 pls remember that u should set a initial length for a map.

Reply:
ok

Reply:
fix